miqa/learning/nn_classifier.py

```diff
@@ -302,6 +301,47 @@ def evaluate1(model, image_path):
     return labeled_results
 
 
+def evaluateMany(model, image_paths):
+    itk_reader = monai.data.ITKReader()
+    # Define transforms for image
+    train_transforms = Compose(
+        [
+            LoadImaged(keys=['img'], reader=itk_reader),
+            EnsureChannelFirstd(keys=['img']),
+            ScaleIntensityd(keys=['img']),
+            ToTensord(keys=['img']),
+        ]
+    )
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+    results = {}
+    for image_path in image_paths:
+        evaluation_ds = monai.data.Dataset(
+            data=[
```

Comment:
You should construct `data` in a for loop, then do `evaluation_ds = monai.data.Dataset(` and the rest of this just once.

Reply:
cool, I didn't know we could do that. Is this second attempt what you meant?